native-activity/app/src/main/cpp/main.cpp

```diff
@@ -138,169 +145,68 @@ struct Engine {
-    config = supportedConfigs[0];
-  }
-
-  if (config == nullptr) {
-    LOGW("Unable to initialize EGLConfig");
-    return -1;
-  }
 
-  /* EGL_NATIVE_VISUAL_ID is an attribute of the EGLConfig that is
-   * guaranteed to be accepted by ANativeWindow_setBuffersGeometry().
-   * As soon as we picked a EGLConfig, we can safely reconfigure the
-   * ANativeWindow buffers to match, using EGL_NATIVE_VISUAL_ID. */
-  eglGetConfigAttrib(display, config, EGL_NATIVE_VISUAL_ID, &format);
-  surface =
-      eglCreateWindowSurface(display, config, engine->app->window, nullptr);
-
-  /* A version of OpenGL has not been specified here.  This will default to
-   * OpenGL 1.0.  You will need to change this if you want to use the newer
-   * features of OpenGL like shaders. */
-  context = eglCreateContext(display, config, nullptr, nullptr);
-
-  if (eglMakeCurrent(display, surface, surface, context) == EGL_FALSE) {
-    LOGW("Unable to eglMakeCurrent");
-    return -1;
-  }
-
-  eglQuerySurface(display, surface, EGL_WIDTH, &w);
-  eglQuerySurface(display, surface, EGL_HEIGHT, &h);
-
-  engine->display = display;
-  engine->context = context;
-  engine->surface = surface;
-  engine->width = w;
-  engine->height = h;
-  engine->state.angle = 0;
-
-  // Check openGL on the system
-  auto opengl_info = {GL_VENDOR, GL_RENDERER, GL_VERSION, GL_EXTENSIONS};
-  for (auto name : opengl_info) {
-    auto info = glGetString(name);
-    LOGI("OpenGL Info: %s", info);
-  }
-  // Initialize GL state.
-  glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_FASTEST);
-  glEnable(GL_CULL_FACE);
-  glShadeModel(GL_SMOOTH);
-  glDisable(GL_DEPTH_TEST);
-
-  return 0;
-}
-
-/**
- * Tear down the EGL context currently associated with the display.
- */
-static void engine_term_display(Engine* engine) {
-  if (engine->display != EGL_NO_DISPLAY) {
-    eglMakeCurrent(engine->display, EGL_NO_SURFACE, EGL_NO_SURFACE,
-                   EGL_NO_CONTEXT);
-    if (engine->context != EGL_NO_CONTEXT) {
-      eglDestroyContext(engine->display, engine->context);
+    if (!window_initialized) {
+      // If for some reason we were not able to initialize the window geometry,
+      // then we can't assume the buffer format. We could detect the buffer's
+      // format and adjust our buffer fill here to accommodate that, but's a bit
+      // beyond the scope of this sample.
+      return;
     }
-    if (engine->surface != EGL_NO_SURFACE) {
-      eglDestroySurface(engine->display, engine->surface);
+
+    for (auto y = 0; y < buffer.height; y++) {
+      for (auto x = 0; x < buffer.width; x++) {
+        size_t pixel_idx = y * buffer.stride + x;
+        reinterpret_cast<uint32_t*>(buffer.bits)[pixel_idx] =
+            static_cast<uint32_t>(color_);
+      }
     }
-    eglTerminate(engine->display);
+
+    ANativeWindow_unlockAndPost(app->window);
   }
-  engine->Pause();
-  engine->display = EGL_NO_DISPLAY;
-  engine->context = EGL_NO_CONTEXT;
-  engine->surface = EGL_NO_SURFACE;
-}
+};
```

Comment:
stray `;`?

Reply:
One of us is misreading the diff, and I think it's you. This is the end of the class definition. The diff makes it look like a stray `;` at the end of a function because it's not-so-helpfully making this look like an edit to completely unrelated code.